### PR TITLE
Rename OpenDevin to OpenHands

### PR DESCRIPTION
**Pull request by rbren**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**
We are renaming the repo!

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

Here are the commands I ran (using my custom `replaceall` util)

```
replaceall ./ 's/opendevin/openhands/g'
replaceall ./ 's/OpenDevin/OpenHands/g'
replaceall ./ 's/open devin/OpenHands/g'
replaceall ./ 's/github.com\/OpenHands/github.com\/All-Hands-AI/g'
replaceall ./ 's/OpenHands\/OpenHands/All-Hands-AI\/OpenHands/g'
replaceall ./ 's/openhands\/openhands/All-Hands-AI\/OpenHands/g'
replaceall ./ 's/cd OpenHands/cd openhands/g'
replaceall 's/ghcr.io\/All-Hands-AI\/OpenHands/ghcr.io\/all-hands-ai\/openhands'
replaceall ./ 's/ghcr.io\/All-Hands-AI\/OpenHands/ghcr.io\/all-hands-ai\/openhands'
replaceall ./ 's/ghcr.io\/All-Hands-AI\/OpenHands/ghcr.io\/all-hands-ai\/openhands/g'
replaceall ./ 's/DOCKER_ORG=openhands/DOCKER_ORG=all-hands-ai/g'
replaceall ./ 's/organizationName: "Open Hands"/organizationName: "All-Hands-AI"/g'
replaceall ./ 's/organizationName: "OpenHands"/organizationName: "All-Hands-AI"/g'
replaceall ./ 's/OPENDEVIN/OPENHANDS/g'
replaceall ./ 's/open-devin/openhands/g'
replaceall ./ 's/devin/openhands/g'
replaceall ./ 's/DEVIN/OPENHANDS/g'
replaceall ./ 's/OPEN_OPENHANDS/OPENHANDS/g'

# Revert image references in documentation for now
replaceall README.md 's/ghcr.io\/all-hands-ai\/openhands/ghcr.io\/opendevin\/opendevin/g'
replaceall docs/ 's/ghcr.io\/all-hands-ai\/openhands/ghcr.io\/opendevin\/opendevin/g'
```

---
**Other references**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -38,8 +38,8 @@ Set the official repository as your [upstream](https://www.atlassian.com/git/tut
 Add the original repository as upstream:
 
 ```shell
-cd OpenDevin
-git remote add upstream git@github.com:OpenDevin/OpenDevin.git
+cd openhands
```
Comment (xingyaoww) [inline CONTRIBUTING.md:41]:
```suggestion
cd OpenHands
```

**Review by neubig — APPROVED** [review]
This is basically a global search and replace. My suggestion would be that we get this in, and then make any additional fixes on top of this.

File: README.md [hunk 2]
```diff
@@ -66,89 +66,89 @@ docker run -it \
...
-For a list of open source projects and licenses used in OpenDevin, please see our [CREDITS.md](./CREDITS.md) file.
+For a list of open source projects and licenses used in OpenHands, please see our [CREDITS.md](./CREDITS.md) file.
 
 ## 📚 Cite
 
 ```
-@misc{opendevin,
-      title={{OpenDevin: An Open Platform for AI Software Developers as Generalist Agents}},
+@misc{openhands,
+      title={{OpenHands: An Open Platform for AI Software Developers as Generalist Agents}},
```
Comment (xingyaoww) [inline README.md:151]:
```suggestion
      title={{OpenDevin: An Open Platform for AI Software Developers as Generalist Agents}},
```

File: README.md [hunk 3]
```diff
@@ -66,89 +66,89 @@ docker run -it \
...
 
-For a list of open source projects and licenses used in OpenDevin, please see our [CREDITS.md](./CREDITS.md) file.
+For a list of open source projects and licenses used in OpenHands, please see our [CREDITS.md](./CREDITS.md) file.
 
 ## 📚 Cite
 
 ```
-@misc{opendevin,
-      title={{OpenDevin: An Open Platform for AI Software Developers as Generalist Agents}},
+@misc{openhands,
```
Comment (xingyaoww) [inline README.md:150]:
```suggestion
@misc{opendevin,
```

File: containers/app/Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-ARG OPEN_DEVIN_BUILD_VERSION=dev
+ARG OPEN_OPENHANDS_BUILD_VERSION=dev
```
Comment (xingyaoww) [inline containers/app/Dockerfile:1]: We probably need to global rename this to `OPENHANDS_BUILD_VERSION`

**Review by xingyaoww — APPROVED** [review]

File: containers/app/Dockerfile (hunk 4 above)
Comment (xingyaoww) [reply containers/app/Dockerfile:1]: Let's do this in subsequent PR

Comment (rbren) [reply containers/app/Dockerfile]: Fixed :)

**Review by tofarr — APPROVED** [review]

**Pull request merged**